.github/pull_request_template.md

```diff
@@ -0,0 +1,5 @@
+**Authors and Reviewers:** please make sure this pull request complies as much
+as possible with the
+[Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/pulls.html#the-iris-check-list).
+
+---
```

Comment:
@trexfeathers Nice! I was thinking along the lines of something like this...


## 🚀 Pull Request

### Description
<!-- Provide a clear description about your awesome pull request -->
<!-- Tell us all about your new feature, improvement, or bug fix -->



### Additional information
<!-- Provide any further information to help the reviewer understand -->
<details>
<summary>Click to expand <b>this section...</b></summary>

```
Add additional verbose information in a collapsible section.
```
See [here](https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab) for further details.
</details>

---
[Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/pulls.html#the-iris-check-list)

Reply:
What do you think?

Reply:
I'm not sure what problems this solves. We don't have PR categories so the title is purely aesthetic, and it's pretty rare for a PR to be so detailed that it needs headings/collapsing. Indeed those 5% come from devs who probably know these features already. 

I'm worried we're just going to encourage a load of markdown 'artifacts' on the 95% of the PR's that are small enough to be better without them. Are you keen for/expecting PR's to get more detailed in future? 

Now I've seen it I quite like the idea of _only_ including the checklist link in the way you have written it.

Reply:
I personally find this pretty minimal myself. If anything you could drop the `Additional information` section... I think you're right, it would most likely be "just deleted" by most PR authors.

However, what I am keen that we do is have is a consistent branding on GitHub for `iris`, and as such this template suggestion is pretty much in line with the new [Issue templates](https://github.com/SciTools/iris/issues/new/choose).

So, we could trim it down to simply...

## 🚀 Pull Request

### Description
<!-- Provide a clear description about your awesome pull request -->
<!-- Tell us all about your new feature, improvement, or bug fix -->


---
[Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/pulls.html#the-iris-check-list)

Reply:
BTW I think that the actual [Iris pull request check list](https://scitools-iris.readthedocs.io/en/latest/developers_guide/pulls.html#the-iris-check-list) needs a bit of work... but @tkknight may have that in hand :thinking:

Reply:
If consistency is the key factor then I appreciate where you're coming from. I'll make the change either today or Monday

> BTW I think that the actual Iris pull request check list needs a bit of work...

Is that even after [the changes to the checklist that this PR suggests](https://github.com/SciTools/iris/pull/3838/commits/15ebb9965d6552900b7fd881bfcb3b729681dd5e#diff-8cd7796992d8299a5dbab9c6080157eb)?

Reply:
Oops, sorry... Hasn't got that far 🤭

Reply:
FYI, the **developers guide** overhaul now has a branch in my forked iris repo.  It still needs some work.  The PR checklist has been improved I can **rebase** if it is updated in @trexfeathers pr.

- [My branch](https://github.com/SciTools/iris/compare/master...tkknight:developer_guide_overhaul)
- [My branch pr checklist](https://github.com/SciTools/iris/compare/master...tkknight:developer_guide_overhaul#diff-107170be3096da8087bc84d4a3018b83)

Reply:
@bjlittle 08ee8ba8fb32244e7c566138dfae0b7beb4f801d

Reply:
Maybe "Description" could be replaced by "What's new" and the content of that section could then be used to compose the complete "What's new" file when making a new release by combining all the entries from merged pull requests since the previous release, as also proposed here https://github.com/SciTools/iris/issues/3837#issuecomment-691956633? To keep it easy to use, harvesting entries and combining them into a latest.rst file would then be done by a script using the GitHub API.

Reply:
Totally 👍 on this (https://github.com/SciTools/iris/issues/3837#issuecomment-691980738), but I'm very reluctant to introduce something new to the docs build at such a late stage for v3.0.0

Reply:
It's up to you, you know your own timeline best.